Starsim bfc tests

klendathu2k opened the pull request:
Add three tests which exercise starsim via St_geant_Maker.  Duration of the tests is quick O(2min).

klendathu2k committed:
Define three simple simulation tests running starsim via St_geant_Maker.

klendathu2k committed:
Simulation jobs take no input files.

plexoos commented on tests/joblist.json:
```diff
@@ -1123,5 +1123,33 @@
+        "outpath": "",
+        "nevents": "50"
     }
+
```
```suggestion
```

plexoos commented on tests/executest.py:
```diff
@@ -40,7 +40,11 @@ def _json_object_hook(d):
+            if job.inpfile=="":
+                print("root4star -b -q -l \'bfc.C({nevents}, \"{chopts}\")\'".format(**job._asdict()))
+            else:
+                print("root4star -b -q -l \'bfc.C({nevents}, \"{chopts}\", \"{inppath}/{inpfile}\")\'".format(**job._asdict()))                
```
```suggestion
        if args.command:
            if job.inpfile:
                print("root4star -b -q -l \'bfc.C({nevents}, \"{chopts}\", \"{inppath}/{inpfile}\")\'".format(**job._asdict()))
            else:
                print("root4star -b -q -l \'bfc.C({nevents}, \"{chopts}\")\'".format(**job._asdict()))
```

plexoos replied:
Do these minor suggestion look good? If so, let's accept and merge the PR

klendathu2k replied:
> Do these minor suggestion look good? If so, let's accept and merge the PR

Yes, they look good to me.

klendathu2k committed:
Remove whitespace

klendathu2k committed:
Merge branch 'main' into starsim-bfc-tests

plexoos committed:
Update tests/executest.py

plexoos commented:
In order to include these test in CI we need to:
1. add them to the workflow https://github.com/star-bnl/star-sw/blob/c677c84babc7b338ee32805d3f3e67a56690adc4/.github/workflows/build-pull-request.yml#L50
2. provide the generator libraries from https://github.com/star-bnl/star-mcgen

plexoos committed:
add starsim test to CI just to see how they fail...

plexoos commented:
Wait.. what? They did not fail... but how can this `bfc.C(50, "sdt20230214 pythia")` work without Pythia?
The output is likely all empty events:
```
 QA :INFO  - QAInfo: Done with Event [no. 48/run 1/evt. 48/Date.Time 20230214.0/sta 3] Real Time =       0.00 seconds Cpu Time =        0.00 seconds
QA :INFO  - QAInfo: Cpu Times: user / system / user children / system children =     0.00 /     0.00 /     0.00 /     0.00 seconds (tick =   100.00, cps = 1000000)
QA :INFO  - QAInfo: Cpu Times: all threads =     0.000182500 seconds
St_geant_Maker:INFO  - *** St_geant_Maker::Make() == StERR(3) ***
QA :INFO  - QAInfo: Done with Event [no. 49/run 1/evt. 49/Date.Time 20230214.0/sta 3] Real Time =       0.00 seconds Cpu Time =        0.00 seconds
QA :INFO  - QAInfo: Cpu Times: user / system / user children / system children =     0.00 /     0.00 /     0.00 /     0.00 seconds (tick =   100.00, cps = 1000000)
QA :INFO  - QAInfo: Cpu Times: all threads =     0.000164600 seconds
St_geant_Maker:INFO  - *** St_geant_Maker::Make() == StERR(3) ***
QA :INFO  - QAInfo: Done with Event [no. 50/run 1/evt. 50/Date.Time 20230214.0/sta 3] Real Time =       0.00 seconds Cpu Time =        0.00 seconds
QA :INFO  - QAInfo: Cpu Times: user / system / user children / system children =     0.00 /     0.00 /     0.00 /     0.00 seconds (tick =   100.00, cps = 1000000)
QA :INFO  - QAInfo: Cpu Times: all threads =     0.000176900 seconds
QA :INFO  - QAInfo:EventLoop completed code 3
```

`StERR(3)` does not mean anything good I assume

plexoos commented:
```
QA :INFO  - QAInfo:St_geant_Maker       built Jul 28 2023 17:47:16 from Tag $Name:  $ $Id: St_geant_Maker.h,v 1.60 2020/02/26 21:26:20 jwebb Exp $ 
 calling aginit 
 ******************************************************
 * Starting  starsim NwGEANT=  10000000 NwPAW=       0 *
 **********************pid=    69**********************
 ******************************************************
 *                                                    *
 *            W E L C O M E    to   starsim           *
 *                                                    *
 *        Version INITIAL     28 July 2023            *
 *                                                    *
 ******************************************************
 Version 1.29/04 of HIGZ started
 CSINIT: CERNLIB_STDUNIX
 CSINIT: CERNLIB_UNIX
1*****  GEANT Version  3.21/08 Released on 230697
0*****  Correction Cradle Version  0.0800
 *****  RZMAKE. OLD RZ format selected for RZDOC               
  Events will not be splitted into subevents 
StChain:INFO  - TAttr::Found : pythia = 1

 *** Unknown routine or function ---> bpythia
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: MDCY
 *** Unknown command: frame
 *** Unknown command: beam
 *** Unknown command: ener
 *** Unknown routine or function ---> PyTUNE
 *** Unknown routine or function ---> pystat
```

klendathu2k commented:
Just coming back to this topic.  The point of this pull request is to test the frameworks for event generation, rather than testing individual event generators.  So in my next commit I will remove the falsely-passing tests and commit the test that I am really interested in... a test of StarGenerator.  That test will ensure that I don't make any code-breaking commits as I integrate the geant4 maker.

klendathu2k committed:
Merge branch 'star-bnl:main' into starsim-bfc-tests

klendathu2k committed:
Remove tests which fail b/c mcgen not compiled during CI testing

klendathu2k committed:
Remove filter dependence from stargen, and define a stargen.filt bfc option to load it separately

klendathu2k committed:
Setup the StarKinematics event generator to throw particles flat in pt,
eta and phi according to attributes set on the maker.  Add option to BFC
to throw 80 muons flat in pt, eta and phi.  The intent is to provide an
integration test of the StarGenerator framework.

---

Throw particles flat in phase space if appropriate attributes set on the generator.

Add option to throw a flat distribution of particles using StarGenerator, analagous to the "gstar" option.

On genmuons, instantiate the StarPrimaryMaker.  Add kinematics generator.  Setup for 80 muons flat in pt,eta and phi.

Add CI test for StarGenerator

klendathu2k commented:
Note... 1 test fails (128) b/c it is no longer defined (and I am unable to remove the test from the workflow due to lack of permission).

iraklic commented:
There is 128 in `test_id` does corresponding `jobid` need to exist in the json file or that is a placeholder?

plexoos commented:
> Note... 1 test fails (128) b/c it is no longer defined (and I am unable to remove the test from the workflow due to lack of permission).

What lack of permission? You should be able to modify .github/workflows/build-pull-request.yml like anyone else with write permission to this repo. I just checked; job 128 was included because you had it in the initial commit on this branch, 2c7f059b86. However, it was somehow removed in the subsequent commits.

```
    {
        "jobid": "128",
        "descr": "2023 hijing simulation",
        "chopts": "sdt20230214 hijing",
        "inpfile": "",
        "inppath": "",
        "outpath": "",
        "nevents": "50"
    }
```

iraklic commented:
> > Note... 1 test fails (128) b/c it is no longer defined (and I am unable to remove the test from the workflow due to lack of permission).
> 
> What lack of permission? You should be able to modify .github/workflows/build-pull-request.yml like anyone else with write permission to this repo. I just checked; job 128 was included because you had it in the initial commit on this branch, [2c7f059](https://github.com/star-bnl/star-sw/commit/2c7f059b86d36f2fa898117261081398297e9e20). However, it was somehow removed in the subsequent commits.
> 
> ```
>     {
>         "jobid": "128",
>         "descr": "2023 hijing simulation",
>         "chopts": "sdt20230214 hijing",
>         "inpfile": "",
>         "inppath": "",
>         "outpath": "",
>         "nevents": "50"
>     }
> ```

ah, thanks for checking. So @klendathu2k do you want to bring that job back or the commit is ok as it is? Let me know, I can go ahead and approve it
